platform/rct/hal/ADC_stm32/ADC_device_stm32.c

```diff
@@ -76,54 +80,46 @@ int ADC_device_init(void)
         GPIO_InitStructure.GPIO_PuPd = GPIO_PuPd_NOPULL;
         GPIO_Init(GPIOB, &GPIO_InitStructure);
 
-        /* Calibration procedure
-         *  TODO see if this is necessary
-         *
-         ADC_VoltageRegulatorCmd(ADC1, ENABLE);
-
-         ADC_SelectCalibrationMode(ADC1, ADC_CalibrationMode_Single);
-         ADC_StartCalibration(ADC1);
+        /* Calibration procedure */
+        ADC_VoltageRegulatorCmd(ADC3, ENABLE);
 
-         while(ADC_GetCalibrationStatus(ADC1) != RESET );
-         calibration_value = ADC_GetCalibrationValue(ADC1);
-        */
+        /*
+         * Wait at least 10us before starting calibration or enabling.
+         * Since there is no interrupt to tell me this is ready we just
+         * delay one scheduler tick which should be more than 10us.  This
+         * assumes that the scheduler is always active during init.
+         */
+        delayTicks(1);
 
-        ADC_CommonInitTypeDef   ADC_CommonInitStructure;
+        /* We compare against Vref only */
+        ADC_SelectDifferentialMode(ADC3, ADC_Channel_1, DISABLE);
 
-        ADC_CommonInitStructure.ADC_Mode = ADC_Mode_Independent;
-        ADC_CommonInitStructure.ADC_Clock = ADC_Clock_AsynClkMode;
-        ADC_CommonInitStructure.ADC_DMAAccessMode = ADC_DMAAccessMode_Disabled;
-        ADC_CommonInitStructure.ADC_DMAMode = ADC_DMAMode_Circular;
-        ADC_CommonInitStructure.ADC_TwoSamplingDelay = 0;
-        ADC_CommonInit(ADC3, &ADC_CommonInitStructure);
+        /* Now calibrate our ADC */
+        ADC_SelectCalibrationMode(ADC3, ADC_CalibrationMode_Single);
+        ADC_StartCalibration(ADC3);
+        while(SET == ADC_GetCalibrationStatus(ADC3));
```

Comment:
Perhaps it wouldn't be a terrible idea to put a timeout here to prevent a deadlock?